tests/Jackalope/Transport/DoctrineDBAL/ClientTest.php

```diff
@@ -30,6 +30,29 @@ public function testQueryNodes()
         $this->assertEquals(1, count($result->getNodes()));
     }
 
+    /**
+     * If the node was deleted in another session and we refresh the session
+     * what should happen?
+     */
+    public function testRefreshWithDeletedNodes()
+    {
+        $root = $this->session->getNode('/');
+        $article = $root->addNode('article');
+        $article->setProperty('foo', 'bar');
+        $article->setProperty('bar', 'baz');
+        $article->setProperty('bar', 'baz');
+        $this->session->save();
+
+        // simulate node being deleted in another session.
+        $nbRows = $this->getConnection()->exec('DELETE FROM phpcr_nodes WHERE path="/article"');
```

Comment:
can we harden this query by adding the workspace? 

or even better, there are tests in phpcr-api-tests about concurrency using two sessions, maybe you could use that to not bypass the abstractions. then you could move that test to phpcr-api-tests. its an all api behaviour.

there is CombinedManipulationsTest::testRemoveOtherSessionRefreshKeepChanges but it seems no equivalent test for removing in a different session and then refresh discarding changes... looking at CombinedManipulationsTest i feel we should split out a ConcurrencyTest class for all thing with 2 sessions, and probably a RefreshTest for all those refresh things... but i am digressing.

can you do a PR to add the test there, with PRs on both jackalope to see what happens?